presto-main/src/main/java/com/facebook/presto/sql/planner/PlanOptimizers.java

```diff
@@ -433,12 +434,16 @@ public PlanOptimizers(
                         .build()));
 
         if (!forceSingleNode) {
+            builder.add(new ReplicateSemiJoinInDelete()); // Must run before AddExchanges
```

Comment:
`// Must run before DetermineSemiJoinDistributionType`

Reply:
i asked @rschlussel to remove this. Because `ReplicateSemiJoinInDelete` does not really care whether it will run before or after `DetermineSemiJoinDistributionType`. Regardless, it will do its duty.